gfop/get_food_counts.py

```diff
@@ -35,6 +33,31 @@ def get_sample_types(simple_complex='all', level=0):
     return (gfop_metadata[['filename', col_sample_types]].set_index('filename'))
 
 
+def get_unique_features(gnps_network, sample_types):
+    """
+    Filter GNPS network to only features that are unique to individual samples or within a specified group
+    Args:
+        gnps_network (data frame): classical molecular networking with study dataset(s) and reference dataset.
+        sample_types (series): sample group for each reference filename
+    Return:
+        Data frame
+    """
+    # presence of each feature in each group
+    groups = sample_types.unique()
```

Comment:
This should be more concise and more efficient in terms of runtime and memory I think:
```
mask = np.zeros((len(gnps_network, sample_types.nunique()), bool)
for i, (_, filenames) in enumerate(sample_types.reset_index().groupby(sample_types.name)):
    mask[:, i] = gnps_network['UniqueFileSources'].str.contains('|'.join(filenames))    # This is regex OR (|) matching.
return gnps_network.loc[mask.sum(axis=1) == 1]
```

Reply:
When I tested this, I get a table of all False

Reply:
Yeah, sorry, I just wrote it by looking at it without actually testing.

`'|'.join(filenames` -> `'|'.join(filenames['filename']` should work.

(Also a missing parenthesis on the first line for `len`).